src/AspNetCoreModuleV2/AspNetCore/HandlerResolver.cpp

```diff
@@ -17,129 +17,146 @@ const PCWSTR HandlerResolver::s_pwzAspnetcoreOutOfProcessRequestHandlerName = L"
-    if (!GetModuleHandleEx(0, pstrHandlerDllName, &m_hRequestHandlerDll))
+    if (!GetModuleHandleEx(0, pstrHandlerDllName, &hRequestHandlerDll))
     {
-        if (pConfiguration->QueryHostingModel() == APP_HOSTING_MODEL::HOSTING_IN_PROCESS)
+        if (pConfiguration.QueryHostingModel() == APP_HOSTING_MODEL::HOSTING_IN_PROCESS)
         {
             std::unique_ptr<HOSTFXR_OPTIONS> options;
 
             RETURN_IF_FAILED(HOSTFXR_OPTIONS::Create(
                 NULL,
-                pConfiguration->QueryProcessPath()->QueryStr(),
+                pConfiguration.QueryProcessPath().c_str(),
                 pApplication.GetApplicationPhysicalPath(),
-                pConfiguration->QueryArguments()->QueryStr(),
+                pConfiguration.QueryArguments().c_str(),
                 options));
 
-            RETURN_IF_FAILED(location.Copy(options->GetExeLocation()));
+            location = options->GetExeLocation();
 
-            if (FAILED_LOG(hr = FindNativeAssemblyFromHostfxr(options.get(), pstrHandlerDllName, &struFileName)))
+            if (FAILED_LOG(hr = FindNativeAssemblyFromHostfxr(*options, pstrHandlerDllName, handlerDllPath)))
             {
                 EventLog::Error(
                     ASPNETCORE_EVENT_INPROCESS_RH_MISSING,
                     ASPNETCORE_EVENT_INPROCESS_RH_MISSING_MSG,
-                    struFileName.IsEmpty() ? s_pwzAspnetcoreInProcessRequestHandlerName : struFileName.QueryStr());
+                    handlerDllPath.empty() ? s_pwzAspnetcoreInProcessRequestHandlerName : handlerDllPath.c_str());
 
                 return hr;
             }
         }
         else
         {
-            if (FAILED_LOG(hr = FindNativeAssemblyFromGlobalLocation(pConfiguration, pstrHandlerDllName, &struFileName)))
+            if (FAILED_LOG(hr = FindNativeAssemblyFromGlobalLocation(pConfiguration, pstrHandlerDllName, handlerDllPath)))
             {
                 EventLog::Error(
                     ASPNETCORE_EVENT_OUT_OF_PROCESS_RH_MISSING,
                     ASPNETCORE_EVENT_OUT_OF_PROCESS_RH_MISSING_MSG,
-                    struFileName.IsEmpty() ? s_pwzAspnetcoreOutOfProcessRequestHandlerName : struFileName.QueryStr());
+                    handlerDllPath.empty() ? s_pwzAspnetcoreOutOfProcessRequestHandlerName : handlerDllPath.c_str());
 
                 return hr;
             }
         }
 
-        LOG_INFOF("Loading request handler:  %S", struFileName.QueryStr());
+        LOG_INFOF("Loading request handler:  %S", handlerDllPath.c_str());
 
-        m_hRequestHandlerDll = LoadLibraryW(struFileName.QueryStr());
-        RETURN_LAST_ERROR_IF_NULL(m_hRequestHandlerDll);
+        hRequestHandlerDll = LoadLibrary(handlerDllPath.c_str());
+        if (preventUnload)
+        {
+            // Pin module in memory
+            GetModuleHandleEx(GET_MODULE_HANDLE_EX_FLAG_PIN, pstrHandlerDllName, &hRequestHandlerDll);
```

Comment:
What happens when you call FreeModule on hRequestHandlerDll when it is pinned?

Reply:
Nothing